spark_log_parser/parsing_models/application_model_v2.py

```diff
@@ -134,10 +134,14 @@ def getSQLinfo(self, appobj):
             sql_jobs = []
             sql_stages = []
             sql_tasks = []
-            for jid, job in appobj.jobs.items():
 
-                if "end_time" not in sql.keys():
-                    sql["end_time"] = appobj.finish_time
+            # Sometimes an SQL event will be missing. To be informative, both
+            # events must be present. But this information is not critical, so
+            # if either event is missing then simply reject the SQL data
+            if "start_time" not in sql.keys() or "end_time" not in sql.keys():
```

Comment:
Not critical, but just to understand:

Could you do just `if "end_time" not in sql.keys()`. This is on the assumption that if an `end_time` exists, then a `start_time` should as well.

Reply:
This won't work because "start_time" and "end_time" come from separate events in the Spark eventlog. So it's possible for one, or other other, or both to be missing.

Reply:
Ah, I see.